src/extension-support/extension-manager.js

```diff
@@ -124,7 +149,7 @@ class ExtensionManager {
 
             const extension = builtinExtensions[extensionURL]();
             const extensionInstance = new extension(this.runtime);
-            return this._registerInternalExtension(extensionInstance).then(serviceName => {
+            return Promise.resolve(this._registerInternalExtension(extensionInstance)).then(serviceName => {
                 this._loadedExtensions.set(extensionURL, serviceName);
```

Comment:
All this is synchronous now, right? In that case it might be easier to read & understand if it's written this way:
```js
this._registerInternalExtension(...);
this._loadedExtensions.set(...);
return Promise.resolve();
```